test/unit/org/apache/cassandra/gms/GossiperTest.java

```diff
@@ -188,7 +188,7 @@ public void testAssassinatedNodeWillNotContributeToVersionCalculation() throws E
         Gossiper.instance.assassinateEndpoint("127.0.0.4");
 
         assertTrue(Gossiper.instance.endpointStateMap.containsKey(InetAddressAndPort.getByName("127.0.0.4")));
-        assertNull(Gossiper.instance.upgradeFromVersionSupplier.get().value());
+        assertTrue(Gossiper.instance.upgradeFromVersionSupplier.get().canMemoize());
```

Comment:
what is that? is not that something different?

Reply:
Just want to make sure the supplier is getting value which can memoize. The null value is the expected memoized value. So they are basically same.